src/main/java/io/confluent/idesidecar/restapi/cache/KafkaConsumerClients.java

```diff
@@ -0,0 +1,53 @@
+package io.confluent.idesidecar.restapi.cache;
+
+import com.github.benmanes.caffeine.cache.CaffeineSpec;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import java.util.Properties;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.common.serialization.ByteArrayDeserializer;
+
+@ApplicationScoped
+public class KafkaConsumerClients extends Clients<KafkaConsumer<byte[], byte[]>> {
+  // Evict cached Consumer instances after 5 minutes of inactivity
+  private static final String CAFFEINE_SPEC = "expireAfterAccess=5m";
+
+  private static final ByteArrayDeserializer BYTE_ARRAY_DESERIALIZER = new ByteArrayDeserializer();
+
+  @Inject
+  ClusterCache clusterCache;
+
+  public KafkaConsumerClients() {
+    super(CaffeineSpec.parse(CAFFEINE_SPEC));
+  }
+
+  public KafkaConsumer<byte[], byte[]> getClient(
+      String connectionId,
+      String clusterId,
+      Properties configOverrides
+  ) {
+    String cacheKey = clusterId;
+    if (!configOverrides.isEmpty()) {
+      // Cache the consumer by the cluster ID and the hash of the config overrides
+      cacheKey = clusterId + "-" + Integer.toHexString(configOverrides.hashCode());
```

Comment:
> Do we need to be concerned about configOverrides.hashCode() from multiple Properties objects returning the same hash code value?

I'm hoping this doesn't happen. 

> Did you consider just changing the key in Clients to be Object so that any cache key (based upon hashcode and equals) could be used?

Are you thinking that we pass an instance of something like `record KafkaConsumerCacheKey(String clusterId, Properties configOverrides){}`? 

I feel like if we can't guarantee that `configOverrides.hashCode()` will remain same across multiple `Properties` instances, then the same will apply to the `KafkaConsumerCacheKey` instances by extension. No?

That said, I like the `KafkaConsumerCacheKey` approach since it's cleaner and more readable.

Reply:
Consistency is not a problem -- hash codes are always consistent for the same or equivalent objects.

I'm more concerned about clashes, where two different `Properties` objects that are not equal might have the same hash code. I think because we're including the connection ID in the cache key, those two duplicates would have to be from the same connection (at least).

What I don't understand is whether that's a practical problem. Why are we including the client properties in the key in the first place, since we're likely just using the same properties for all clients using that connection? We should be clearing out the client cache for a connection that has changed (via events), so even if we just used the connection ID as a cache key, subsequent requests for that connection (after it is changed) should get new clients.

Why is caching by connection ID (in a cache of a particular kind of client) not sufficient?

Reply:
Also, if we do want to include the client properties in the cache key, why would we not use the client configuration rather than just the configuration overrides?

Reply:
> Why are we including the client properties in the key in the first place, since we're likely just using the same properties for all clients using that connection

Actually, this is not true. The consumer config overrides is directly informed by the consumer request (of which there are currently 4 combinations), see [this method](https://github.com/confluentinc/ide-sidecar/pull/112/files#diff-7cda12ef080575ba4739bf3b6122c7d50aeb67ecab280da08cd15a631acf1d8dR46-R59):

```java
// In src/main/java/io/confluent/idesidecar/restapi/messageviewer/data/SimpleConsumeMultiPartitionRequest.java
  /**
   * Returns the consumer configuration overrides that must be applied when creating a Kafka
   * consumer for this request.
   */
  public Properties consumerConfigOverrides() {
    var props = new Properties();
    if (maxPollRecords() != null) {
      props.put(ConsumerConfig.MAX_POLL_RECORDS_CONFIG, maxPollRecords());
    }
    if (fetchMaxBytes() != null) {
      props.put(ConsumerConfig.FETCH_MAX_BYTES_CONFIG, fetchMaxBytes());
    }
    return props;
  }
```